packages/@aws-cdk/aws-rds/test/test.cluster.ts

```diff
@@ -839,6 +839,140 @@ export = {
     test.done();
   },
 
+  'cluster with s3 import bucket adds supported feature name to IAM role'(test: Test) {
+    // GIVEN
+    const stack = testStack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+
+    const bucket = new s3.Bucket(stack, 'Bucket');
+
+    // WHEN
+    new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.auroraPostgres({
+        version: AuroraPostgresEngineVersion.VER_10_12,
+      }),
+      instances: 1,
+      masterUser: {
+        username: 'admin',
+      },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+      s3ImportBuckets: [bucket],
+    });
+
+    // THEN
+    expect(stack).to(haveResource('AWS::RDS::DBCluster', {
+      AssociatedRoles: [{
+        RoleArn: {
+          'Fn::GetAtt': [
+            'DatabaseS3ImportRole377BC9C0',
+            'Arn',
+          ],
+        },
+        FeatureName: 's3Import',
+      }],
+    }));
+
+    test.done();
+  },
+
+  'throws when s3 import bucket is supplied and database engine does not support it'(test: Test) {
+    // GIVEN
+    const stack = testStack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+
+    const bucket = new s3.Bucket(stack, 'Bucket');
+
+    // WHEN / THEN
+    test.throws(() => {
+      new DatabaseCluster(stack, 'Database', {
+        engine: DatabaseClusterEngine.auroraPostgres({
+          version: AuroraPostgresEngineVersion.VER_10_4,
+        }),
+        instances: 1,
+        masterUser: {
+          username: 'admin',
+        },
+        instanceProps: {
+          instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+          vpc,
+        },
+        s3ImportBuckets: [bucket],
+      });
+    }), /s3Import is not supported for engine version 10.4. Use an engine version that supports the s3Import feature./;
+
+    test.done();
+  },
+
+  'throws when s3 export bucket is supplied and database engine does not support it'(test: Test) {
+    // GIVEN
+    const stack = testStack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+
+    const bucket = new s3.Bucket(stack, 'Bucket');
+
+    // WHEN / THEN
+    test.throws(() => {
+      new DatabaseCluster(stack, 'Database', {
+        engine: DatabaseClusterEngine.auroraPostgres({
+          version: AuroraPostgresEngineVersion.VER_10_4,
+        }),
+        instances: 1,
+        masterUser: {
+          username: 'admin',
+        },
+        instanceProps: {
+          instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+          vpc,
+        },
+        s3ExportBuckets: [bucket],
+      });
+    }), /s3Import is not supported for engine version 10.4. Use an engine version that supports the s3Import feature./;
+
+    test.done();
+  },
+
+  'cluster with s3 export bucket adds supported feature name to IAM role'(test: Test) {
```

Comment:
Maybe one positive test-case with import and export at the same time? I don't think splitting them adds much.

Reply:
fair point, folding them into one test